### ISSUE TITLE
Fails to serialize and deserialize private types

### ISSUE THREAD

**Issue by egracer** [jet/Vertigo.Json#7]
```
type private Person = 
{
    firstName: string
    lastName: string
}

let x = 
{
    firstName = "John"
    lastName = "Doe"
}

Json.serialize x
```

The above snippet throws an error saying that the type is not a Record. It would be great if the error message could identify that the type is inaccessible, or suggest making sure that the type is publicly scoped.

**Comment by Tombert** [comment]
This is definitely something we should look into.  I believe that we have access to that information upon doing reflection, so I do not believe it will be difficult to add.

**Comment by bartelink** [comment]
Yes, friendly messages are good. OOTB, Json.net silently omits the fields, which is definitely to be avoided too ;)
